request_router.h

```diff
@@ -0,0 +1,21 @@
+#ifndef REQUEST_ROUTER_H
+#define REQUEST_ROUTER_H
+
+#include "server_config.h"
+#include <cstddef>
+
+class RequestRouter {
+  public:
+    // TODO: Reduce duplication between this and the handlers
```

Comment:
RequestRouter is currently just a wrapper atop the handlers. In the future, we could take a polymorphic approach where we have a RequestParser that parses a Request into an EchoRequest or a StaticFileRequest. Then, our RequestRouter uses polymorphism to automatically call the right handler. It'd be more modularity and less duplication, at the cost of more complexity.